test/Error.spec.js

```diff
@@ -182,4 +182,37 @@ describe('Error', () => {
 			done()
 		})
 	})
+
+	it('should unmount child components', (done) => {
+		let container = document.createElement('div')
+		let stack = []
+		let error = console.error
+		console.error = () => {}
+
+		render(class {
+			componentWillUnmount() {
+				stack.push(1)
+			}
+			getInitialState() {
+				throw new Error('Error!')
+			}
+			render() {
+				return class {
+					componentWillUnmount() {
+						stack.push(1)
+					}
+					render() {
+						return h('div', 'xxx')
+					}
+				}
+			}
+		}, container)
+
+		nextTick(() => {
+			assert.lengthOf(stack, 1)
```

Comment:
Perhaps I don't understand. Why is this not 2?

Reply:
An error boundary component itself does not unmount, its children(what it renders) do, it could be a `div` or in this case another component. 

This tests that only the child components `componentWillUnmount` is called. 

For example you could do something like this –
 [https://codepen.io/gaearon/pen/wqvxGa?editors=0010](https://codepen.io/gaearon/pen/wqvxGa?editors=0010).

Reply:
Perhaps we need to rethink allowing returning elements from componentDidCatch to better match React and only allow implicit state return.

Reply:
seems reasonable